Lesson18/duygu/script.js

```diff
@@ -0,0 +1,197 @@
+// 4. Write a function that takes an array of integers and returns a new array with odd numbers only.
+// Input: [1, 2, 3, 4, 7, 10]
+// Output: [1, 3, 7]
+let arrayEx = [1, 2, 3, 4, 7, 10];
+
+function getOddNumbers(array) {
+  let oddNumbers = [];
+  for (let i = 0; i < array.length; i++) {
+    if (array[i] % 2 !== 0) {
+      oddNumbers.push(array[i]);
+    }
+  }
+  return oddNumbers;
+}
+console.log("Ex4");
+console.log(getOddNumbers(arrayEx));
+console.log('----------------------------');
+// 5. Write a function that takes an array of integers and returns a new array with even numbers only.
+// Input: [1, 2, 3, 4, 7, 10]
+// Output: [1, 2, 4, 10]
+const testArrayEx5 = [1, 2, 3, 4, 7, 10];
+
+function getEvenNumbers(array) {
+  let evenNumbers = [];
+  for (let i = 0; i < array.length; i++) {
+    if (array[i] % 2 === 0) {
+      evenNumbers.push(array[i]);
+    }
+  }
+  return evenNumbers;
+}
+
+console.log('Exercise 5: ');
+console.log(getEvenNumbers(testArrayEx5));
+console.log('----------------------------');
+// 6. Write a function that takes an array and a value, and returns a new array with all occurrences of the value removed.
+// Input: ([1, 2, 3, 4, 2], 2)
+// Output: [1, 3, 4]
+const testArrayEx6 = [1, 2, 3, 4, 2];
+
+function removeValueFromArray(value, array) {
+  let result = [];
+
+  for (let i = 0; i < array.length; i++) {
+    if (value !== array[i]) {
+      result.push(array[i]);
+    }
+    // result = [...result, array[i]]
+
+  }
+  return result;
+}
+
+console.log("exercise 6:");
+console.log(removeValueFromArray(2, testArrayEx6));
+console.log('----------------------------');
+// 7. Write a function that takes an object and returns the keys as an array.
+// Input: {name: "John", age: 30, city: "New York"}
+// Output: ["name", "age", "city"]
+const testObjEx7 = { name: "John", age: 30, city: "New York" };
+function getObjectKeys(obj) {
+  let keys = [];
+  for (let key in obj) {
+    keys.push(key);
+  }
+  return keys;
+}
+
+console.log("exercise 7:");
+console.log(getObjectKeys(testObjEx7));
+console.log('----------------------------');
+// 8. Write a function that takes an object and returns the values as an array.
+// Input: {name: "John", age: 30, city: "New York"}
+// Output: ["John", 30, "New York"]
+const testObjEx8 = { name: "John", age: 30, city: "New York" };
+function getObjectvalues(obj) {
+  let values = [];
+  for (let key in obj) {
+    values.push(obj[key]);
+  }
+  return values;
+}
+console.log("exercise 8:");
+console.log(getObjectvalues(testObjEx8));
+console.log('----------------------------');
+// 9. Write a function that takes an object and a key, and returns true if the key exists in the object, otherwise false.
+// Input: ({name: "John", age: 30, city: "New York"}, "age")
+// Output: true
+// Input: ({name: "John", age: 30, city: "New York"}, "surname")
+// Output: false
+const testObjEx9 = { name: "John", age: 30, city: "New York" };
+
+function keyExists(obj, key) {
+  return key in obj;
```

Comment:
Built in alternative: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty